wicker/core/storage.py

```diff
@@ -275,24 +275,47 @@ class WickerPathFactory:
                     - _SUCCESS
                     - part-0-attempt-1234.parquet
                     - part-1-attempt-2345.parquet
+
+    When a prefix_replace_path is passed the paths of each location are altered such that
+    the prefix_to_trim is replaced with prefix_replace_path. This is generally done in the case
+    of mounted buckets where the fully-qualified posix path is required.
     """
 
-    def __init__(self, root_path: str, store_concatenated_bytes_files_in_dataset: bool = False) -> None:
+    def __init__(
+        self, root_path: str, prefix_replace_path: str = "", store_concatenated_bytes_files_in_dataset: bool = False
+    ) -> None:
         """Init the path factory.
 
         Object to form the expected paths and return them to the user based of root path and storage bool.
 
-        Args:.
+        Args:
             root_path (str): File system location of the root of the wicker file structure.
+            prefix_replace_path (str, optional): Prefix to replace designated prefixes passed to functions.
             store_concatenated_bytes_files_in_dataset (bool, optional): Whether to assume concatenated bytes files
                 are stored in the dataset folder. Defaults to False
         """
+        self._prefix_replace_path = prefix_replace_path
         self.root_path: str = root_path
         self.store_concatenated_bytes_files_in_dataset = store_concatenated_bytes_files_in_dataset
 
     def __eq__(self, other: Any) -> bool:
         return super().__eq__(other) and type(self) == type(other) and self.root_path == other.root_path
 
+    @staticmethod
+    def __format_pathing(input_path: str, prefix_replace_path: str = "", prefix_to_trim: Optional[str] = None) -> str:
```

Comment:
ditto, should both prefixes adopt the same style of optional or empty?

Reply:
See the above, personal preference based on 335